src/VCR/RequestMatcher.php

```diff
@@ -1,102 +0,0 @@
-<?php
-
-namespace VCR;
-
-/**
- * Collection of matcher methods to match two requests.
- */
-class RequestMatcher
```

Comment:
I added classes for the matchers, but what I put in the availableRequestMatchers member (in the createAvailableMatchers method) are still callables. You can still use addRequestMatcher and enableRequestMatchers as before. I just ran `phpunit ./tests/VCR/ConfigurationTest.php`:

```
There was 1 failure:
1) VCR\ConfigurationTest::testEnableRequestMatchers
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     0 => Array (
-        0 => 'VCR\RequestMatcher'
-        1 => 'matchHeaders'
+        0 => VCR\RequestMatchers\HeadersMatcher Object (...)
+        1 => 'match'
     )
     1 => Array (
-        0 => 'VCR\RequestMatcher'
-        1 => 'matchBody'
+        0 => VCR\RequestMatchers\BodyMatcher Object (...)
+        1 => 'match'
     )
 )
/home/tylercollier/repos/php-vcr-tylercollier/tests/VCR/ConfigurationTest.php:80
```

Simple to fix up that test :-)

Just for own my own double-check, I put this in `ConfigurationTest.php`, and it passes:

```
public function returnTrue() {
    return true;
}

public function testAddRequestMatchersWithObjectAndMethodNameCallable()
{
    $expected = array(&$this, 'returnTrue');
    $this->config->addRequestMatcher('new_matcher', $expected);
    $this->assertContains($expected, $this->config->getRequestMatchers());
}
```

@dbu is correct though that if people had referenced one of the static functions (e.g. matchMethod), then maybe I should put that class back. I would recommend deprecating them, since we have improved replacements.